tasks/install-matlab/v1/src/matlab.ts

```diff
@@ -70,18 +110,39 @@ async function resolveLatest(): Promise<string> {
     });
 }
 
-export async function setupBatch(platform: string): Promise<void> {
-    const batchInstallDir = script.defaultInstallRoot(platform, "matlab-batch");
-    const exitCode = await script.downloadAndRunScript(platform, "https://ssd.mathworks.com/supportfiles/ci/matlab-batch/v0/install.sh", batchInstallDir);
+export async function setupBatch(platform: string, architecture: string) {
+    if (architecture !== "x64") {
+        return Promise.reject(Error(`This action is not supported on ${platform} runners using the ${architecture} architecture.`));
+    }
 
-    if (exitCode !== 0) {
-        return Promise.reject(Error(`Failed to install matlab-batch. Script exited with non-zero code ${exitCode}.`));
+    const matlabBatchRootUrl: string = "https://ssd.mathworks.com/supportfiles/ci/matlab-batch/v1/";
+    let matlabBatchUrl: string;
+    let matlabBatchExt: string = "";
+    switch (platform) {
+        case "win32":
+            matlabBatchExt = ".exe";
+            matlabBatchUrl = matlabBatchRootUrl + "win64/matlab-batch.exe";
+            break;
+        case "linux":
+            matlabBatchUrl = matlabBatchRootUrl + "glnxa64/matlab-batch";
+            break;
+        case "darwin":
+            matlabBatchUrl = matlabBatchRootUrl + "maci64/matlab-batch";
+            break;
+        default:
+            return Promise.reject(Error(`This action is not supported on ${platform} runners.`));
     }
 
+    const tempPath = await downloadToolIfNecessary(matlabBatchUrl, `matlab-batch${matlabBatchExt}`);
+    const matlabBatchPath = await toolLib.cacheFile(tempPath, `matlab-batch${matlabBatchExt}`, "matlab-batch", "1.0.0");

```

Comment:
Yeah, we're never looking it up from the cache, we're only adding it there (replacing it each time if it already existed). We're adding it to the cache because the download goes to runner temp by default which seems to clear between every step. There's not a well documented alternative for where to put things onto hosted runners.